runtime/permissions/lib.rs

```diff
@@ -3514,17 +3502,29 @@ impl PermissionsContainer {
     Ok(())
   }
 
-  /// This checks to see if the allow-all flag was passed, not whether all
-  /// permissions are enabled!
```

Comment:
Seems ok to change this so it's derived. Having separate state for an "all" permission seems more problematic.